odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -0,0 +1,1217 @@
+            'custom_properties': property_definition,
+        })
+        self.assertEqual(
+            message.custom_properties[0]['value'],
+            self.partner_2.id,
+            msg='Should not take the default value',
+        )
+
+    def test_properties_field_read(self):
+        """Test the behavior of the read method.
+
+        In comparison with a simple "record.properties", the read method should not
+        record a recordset for the many2one, but a tuple with the record id and
+        the record name_get.
+        """
+        properties_values = (self.message_1 | self.message_3).read(['custom_properties'])
+
+        self.assertEqual(len(properties_values), 2)
+
+        properties_message_1 = properties_values[0]['custom_properties']
+        properties_message_3 = properties_values[1]['custom_properties']
+
+        self.assertTrue(isinstance(properties_message_1, list))
+        self.assertTrue(isinstance(properties_message_3, list))
+
+        self.assertEqual(len(properties_message_1), 2, msg="Message 1 has 2 properties")
+        self.assertEqual(len(properties_message_3), 1, msg="Message 3 has 1 property")
+
+        self.assertEqual(
+            properties_message_1[0]['id'], 'discussion_color_code',
+            msg='First message 1 property should be "discussion_color_code"')
+        self.assertEqual(
+            properties_message_1[1]['id'], 'moderator_partner_id',
+            msg='Second message 1 property should be "moderator_partner_id"')
+        self.assertEqual(
+            properties_message_3[0]['id'], 'state',
+            msg='First message 3 property should be "state"')
+
+        many2one_property = properties_message_1[1]
+        self.assertEqual(
+            many2one_property['string'], 'Partner',
+            msg='Parent property definition must be present when reading child')
+        self.assertEqual(
+            many2one_property['type'], 'many2one',
+            msg='Parent property definition must be present when reading child')
+        self.assertEqual(
+            many2one_property['model'], 'res.partner',
+            msg='Parent property definition must be present when reading child')
+        self.assertEqual(many2one_property['value'], (self.partner.id, self.partner.display_name))
+
+        # disable the name_get
+        properties_values = (self.message_1 | self.message_3).read(['custom_properties'], load=None)
+        many2one_property = properties_values[0]['custom_properties'][1]
+
+        self.assertEqual(
+            many2one_property['value'], self.partner.id,
+            msg='If name_get is disable, should only return the record id')
+
+    def test_properties_field_many2one_basic(self):
+        """Test the basic (read, write...) of the many2one property."""
+        self.message_2.custom_properties = self._to_list(
+            {'moderator_partner_id': [self.partner_2.id, 'res.partner']},
+            self.message_2)
```

Comment:
Damn, the `_to_list` call hides the good stuff.  I would like to see the explicit value you should use in an assignment.